File: bin/parse.js
const fs = require('fs');
const path = require('path');
const request = require('request');
const fetch = require('node-fetch');
const inquirer = require('inquirer');

inquirer.prompt([
  {
    type: 'input',
    message: '请输入你要去水印的链接：',
    name: 'url',
    default: 'https://v.douyin.com/Jajqoxv/'
  }
]).then(answer => {
  parseDouyinUrl(answer.url)
});

/**
 * 解析抖音链接，并去除水印
 * @param {string} url 解析地址
 */
function parseDouyinUrl(url) {
  fetch(url).then(res => {
    const [, item_ids] = /video\/(\d+)\//.exec(res.url)
    fetch(`https://www.iesdouyin.com/web/api/v2/aweme/iteminfo/?item_ids=${item_ids}`).then(res => res.text()).then(res => {
      const resUrl = JSON.parse(res).item_list[0].video.play_addr.url_list[0].replace('playwm', 'play')
      const filename = JSON.parse(res).item_list[0].desc + '.mp4';
      downloadParseUrl(resUrl, filename, () => {
        console.log('文件位置：download/'+filename + '.mp4\n下载完毕')
      })
    }).catch(err => {
      console.log(`输入内容“${url}”解析失败：`, err);
    })
  }).catch(err => {
    console.log(`输入内容“${url}”解析失败：`, err);
  })
}

/**
 * 下载链接生成的视频
 *
 * @param {string} url 下载地址
 * @param {string} filename  下载到本地的名字
 * @param {function} callback  回调
 */
function downloadParseUrl(url, filename, callback) {
  var stream = fs.createWriteStream( path.join(__dirname, '../download/' + filename));
  request(url).pipe(stream).on('close', callback);
}

File: bin/parse.py
#encoding=utf-8
import requests, re, json
url = input("请输入你要去水印的链接：(eg: https://v.douyin.com/Jajqoxv/)\n")
url = url if url!='' else 'https://v.douyin.com/Jajqoxv/'
print('获取item_ids中...')
est1 = requests.get(url)
uq = re.findall('video/(\d+)/',str(est1.url))[0]
ur11 = f'https://www.iesdouyin.com/web/api/v2/aweme/iteminfo/?item_ids={uq}'
print('获取url中...')
est2 = requests.get(ur11).text
js1 = json.loads(est2)
url3 = str(js1['item_list'][0]['video']['play_addr']['url_list'][0]).replace('playwm','play')
title = str(js1['item_list'][0]['desc'])
headers = {'User-Agent': 'Mozilla/5.0 (Linux; Android 5.0; SM-G900P Build/LRX21T) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/66.0.3329.0 Mobile Safari/537.36'}
print('下载中...')
est = requests.get(url=url3,headers=headers)
with open(f'download/{title}.mp4','wb') as f:
    f.write(est.content)
    print('文件位置：download/'+title + '.mp4\n下载完毕')
